### PR TITLE
Changelog for `v2.8.0rc2`

### PR DESCRIPTION
**Pull request by remi-kazeroni**
## Description

Changelog for `v2.8.0rc2`

Related to #1951 

Link to documentation: https://esmvaltool--1959.org.readthedocs.build/projects/ESMValCore/en/1959/changelog.html

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
~~- [ ] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added~~
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
~~- [ ] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly~~
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
doc test failing because of this `/home/docs/checkouts/readthedocs.org/user_builds/esmvalcore/checkouts/1959/doc/changelog.rst:: ERROR: Anonymous hyperlink mismatch: 1 references but 0 targets.`

**Comment by remi-kazeroni** [comment]
Yeah, I did some copy-pasting in the last commit without formatting all the links properly. Will fix that now and make this rfr. We may need to give the TLT the chance to take a look. Will post on it asap.

**Review by valeriupredoi — APPROVED** [review]
looks good to me, bud! Cheers very much for doing this :beer:

**Review by schlunma — APPROVED** [review]
Thanks @remi-kazeroni!

File: doc/changelog.rst [hunk 1]
```diff
@@ -12,25 +12,31 @@ v2.8.0
 Highlights
 ~~~~~~~~~~
 
+-  ESMValCore now supports wildcards in recipes and offers improved support for ancillary variables and dataset versioning thanks to contributions by `Bouwe Andela <https://github.com/bouweandela>`__. For details, see :ref:`Automatically populating a recipe with all available datasets <dataset_wildcards>` and :ref:`Defining supplementary variables <supplementary_variables>`. 
+-  Support for CORDEX datasets has been added by `sloosvel <https://github.com/sloosvel>`__.
```
Comment (sloosvel) [inline doc/changelog.rst:16]:
```suggestion
-  Support for CORDEX datasets in a rotated pole coordinate system has been added by `sloosvel <https://github.com/sloosvel>`__.
```

**Comment by bouweandela** [comment]
Thank you for creating the changelog @remi-kazeroni. Is it OK if I push the changes directly to this branch or shall I make a pull request?

**Comment by remi-kazeroni** [comment]
@bouweandela feel free to push the changes here directly. Then I'll take a look. Unless these are big changes, I hope we can agree on merging this PR today and then I could still do the release of v2.8.0rc2 on GitHub (and PyPi)

**Comment by bouweandela** [comment]
I noticed that the `Preprocessor` section has gone missing but the pull requests do have the right label, maybe this is something in the script you used? I added it in the last commit.

**Comment by remi-kazeroni** [comment]
Thanks for your help and suggestions @bouweandela! Looks good to me at first sight but we need to get the build fixed. Large commits and trouble to build locally can make it challenging to debug, spent my afternoon on that yesterday already 😅  

> I noticed that the `Preprocessor` section has gone missing but the pull requests do have the right label, maybe this is something in the script you used? I added it in the last commit.

When I ran `draft_release_notes.py`, I got an empty list for the `Preprocessor` section because not many PRs used that label and even if it was used, the script picks another label which comes before in the alphabetical order.

I agree with your idea of making links to refer to PRs more than once. It's very challenging not be able to cite a PR more than once when you want to highlight it and at the same time provide some guidance to users on how to handle deprecations and co.

Is there any particular reason for not ordering the PRs in ascending PR numbers in every section any more? This is done automatically with `draft_release_notes.py`

**Comment by remi-kazeroni** [comment]
Just some thoughts for the future: I think we should really revise the way we generate the changelog. This requires too much last-minute tweaking for the RM and too many manual steps (probably more than for testing all recipes) to handle 50+ PRs with 2 or 3 labels each. What about having a changelog PR open as soon as we start a new release cycle? We could make it the responsibility of the PR merger or author to add the merged PR to the appropriate section of the changelog. They may know better where to put them than the RM. Before generating the final version of the changelog, the RM would run the script `draft_release_notes.py` only to check that all PRs were added. And then the RM would handle the highlights, messages to the users.

**Comment by valeriupredoi** [comment]
> Just some thoughts for the future: I think we should really revise the way we generate the changelog. This requires too much last-minute tweaking for the RM and too many manual steps (probably more than for testing all recipes) to handle 50+ PRs with 2 or 3 labels each. What about having a changelog PR open as soon as we start a new release cycle? We could make it the responsibility of the PR merger or author to add the merged PR to the appropriate section of the changelog. They may know better where to put them than the RM. Before generating the final version of the changelog, the RM would run the script `draft_release_notes.py` only to check that all PRs were added. And then the RM would handle the highlights, messages to the users.

I agree with @remi-kazeroni - I'd like this discussed on a TLT call - Remi, I'll open an issue about this if that's OK with you!

**Comment by bouweandela** [comment]
The build fails because this `undefined label: esmvaltool:read_icon`, maybe I got the reference wrong?

**Comment by valeriupredoi** [comment]
@bouweandela I don't see the warning you mention at https://readthedocs.org/projects/esmvalcore/builds/19816569/ - all I see is the pandoc warning which I fixed by pinning sphinx in #1976

**Comment by valeriupredoi** [comment]
ah nevermind - it was out of the screen - indeed - that's another warning

**Comment by valeriupredoi** [comment]
excellent! So we don't really really need #1976 for RTD builds - we may need it for local builds since Remi had issues with his - bet let's do it after the release. At any rate - let's get this in, @bouweandela would you care to merge, please, bud? I can do it if you don't do it within a lap at Le Mans/La Sarthe time (~4min) :grin:

**Comment by remi-kazeroni** [comment]
Let me take a very final look now please 👍  And yes, I would have saved a lot of time if I knew that when you click on "rdt: details" > "view raw" you actually see the full log, not the raw .rst file... 🤯 At least I learnt something!

**Comment by bouweandela** [comment]
Whoops, too quick! Sorry @remi-kazeroni

**Comment by valeriupredoi** [comment]
> do it like they do it

on the Discovery Channel :rofl: 

Cheers guys, we are OK for now. Let's get some of these things sorted for 2.9 - at least we have now sorted the many steps for Tool in that dox PR - baby steps

**Pull request merged**